### ISSUE TITLE
Add Surface Deform support for the Physics Panel

### ISSUE THREAD

**Issue by mokujinh** [Mustard2/MustardUI#197]
Hey there, after a few hours of struggling with having more than one physics cage detected by body meshes 
(not 100% related to the UI but it can lead to an improvement). 
So the default "mesh deform" modifier has many issues, in my case having breasts and hips separate cages made it so the hips cage wouldn't work on the body no matter what. 
After a lot of testing and some people helping me out we figured that is just the modifier itself is very buggy and non responsive. 

Now days add-ons with cage based physics mostly just use the "surface deform" modifier, (an example could be the "Simplicage" add-on) it also requires the same amount of setting up except that you just need to add a vertex group for the target mesh to select an area that will react to the modifier specifically. it also allows you to control the strength of the effect the modifier will have on the mesh, which honestly targets the mesh deform modifier obsolete.
Which makes me wonder why the UI requires us to use such an outdated modifier when the surface deform seems to be a better option overall?

Here's an example (provided by @ariespossession user on X) of the difference in the settings between the mesh deform modifier used by the UI and the surface deform modifier made by this user.
![UQ6miWAH](https://github.com/user-attachments/assets/e9cebba8-6b7f-46e7-9e0e-17b576c1f988)
An idea of even having a slider in the physics panel for the "strength" could make a huge difference since people could keyframe this value and save a lot of time in many situations even after baking physics.

My suggestion is to mark the mesh deform modifier obsolete (or have an option to chose which one we could use for the UI) and move on to the surface deform modifier. By now is less buggy and more dependable. The UI configuration for setting up the physics for it and to bind it to all sort of outfits would be extremely useful. Thanks for reading and always providing updates for this amazing tool!

**Comment by Mustard2** [comment]
Yes in fact in the new version I'm coding, you will be able to choose which version of the UI you like.

I know it has been a while, but I'm still working on the whole physics part. Essentially it will be a complete rewrite of the whole thing.

**Comment by Mustard2** [comment]
The UI will now check for both Mesh and Deform modifiers when enabling/disabling physics.
